### PR TITLE
[Routing] Support for Array parameter in Route requirement

### PR DESCRIPTION
**Pull request by raziel057**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

Sometimes it can be usefull to restrict the parameters of a route to one value of a list defined in the application parameters. 

For example, if I take the example provided here:
http://symfony.com/doc/current/book/translation.html#book-translation-locale-url

If I have an `active_languages: [fr, en]` parameter defined in my parameters, it would be useful to be able to provide it directly in a route requirements, like this, to avoid to create another parameter as a string `active_languages_str: "fr|en"`:

``` php
/**
 * @Route("/list/{_locale}", name="list", requirements={"_locale" = "%active_languages%"})
```

I think this case can be applied for other common cases. I can create some tests if this PR is accepted.

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Routing/Router.php [hunk 1]
```diff
@@ -154,6 +155,8 @@ private function resolve($value)
 
             if (is_string($resolved) || is_numeric($resolved)) {
                 return (string) $resolved;
+            } elseif ($acceptArrayParameters && is_array($resolved)) {
```
Comment (derrabus) [inline src/Symfony/Bundle/FrameworkBundle/Routing/Router.php]: You did not import `$acceptArrayParameters` to the variable scope of the anonymous function. This expression will always evaluate to false and throw a notice.

Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Routing/Router.php]: this should be a ``if``, not a ``elseif``, as the previous if returns.

**Comment by stof** [comment]
I see 2 issues in your PR:

- it does not have test covering this feature, meaning it could disappear in a future version without anyone noticing it (until you would deploy a new impacted Symfony release in your own project relying on it). Preventing regressions is the main goal of the testsuite
- your replacement with an imploded works fine only if the parameter is the full requirement. but this is not what the regex replacement parameters is doing. It accepts replacing things like ``foo%locales%``. In such case, your imploded string would lead to a totally different behavior than expected.

**Comment by derrabus** [comment]
I personally would like to see some test to have an actual prove of concept for your feature. Also, please add the checklist that is described [here](http://symfony.com/doc/current/contributing/code/patches.html#make-a-pull-request) to the description of your PR.

Status: Needs Work

**Comment by raziel057** [comment]
@derrabus I've wrote a unit test to cover the new feature.

@stof I don't find a simple way to prevent this bad usage. Is this really necessary?

**Comment by stof** [comment]
Looks like you messed your rebasing

**Pull request closed without merging**